crates/bevy_ecs/src/world/entity_access/entity_mut.rs

```diff
@@ -591,6 +591,74 @@ impl<'w> EntityMut<'w> {
         unsafe { component_ids.fetch_mut_assume_mutable(self.cell) }
     }
 
+    /// Gets the "target" entity of this entity via the [`Relationship`] component with the given [`ComponentId`].
+    ///
+    /// **You should prefer to use the typed API where possible and only
+    /// use this in cases where the actual component types are not known at
+    /// compile time.**
+    ///
+    /// # Examples
+    ///
+    /// ## Single [`ComponentId`]
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # let mut world = World::new();
+    /// let parent = world.spawn_empty().id();
+    /// let child = world.spawn(ChildOf(parent)).id();
+    ///
+    /// // Grab the component ID for `ChildOf` in whatever way you like.
+    /// let component_id = world.register_component::<ChildOf>();
+    ///
+    /// // Then, get the target entity via the 'ChildOf' relationship by ID.
+    /// let target = world.entity(child).get_relationship_by_id(component_id).unwrap();
+    /// assert_eq!(target, parent);
+    /// ```
+    ///
+    /// [`Relationship`]: crate::relationship::Relationship
+    pub fn get_relationship_by_id(&self, relationship_id: ComponentId) -> Option<Entity> {
+        // SAFETY: We have read-only access to all components of this entity.
+        unsafe { self.cell.get_relationship_by_id(relationship_id) }
```

Comment:
And I'd recommend writing these in terms of `as_readonly()`, similar to `EntityMut::get()`, since that means we can write them with safe code and only have to get the lifetimes right in on place: 

```suggestion
        self.as_readonly().get_relationship_by_id(relationship_id)
```